framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -691,95 +771,172 @@ private Range calculateRangeUnaryOp(
             Node leftNode,
             Node rightNode,
             ComparisonOperators op,
-            TransferInput<CFValue, CFStore> p) {
-        List<Boolean> resultValues = new ArrayList<>();
-        if (!isIntRange(leftNode, p) && !isIntRange(rightNode, p)) {
+            TransferInput<CFValue, CFStore> p,
+            CFStore thenStore,
+            CFStore elseStore) {
+        if (isIntRange(leftNode, p) || isIntRange(rightNode, p)) {
             // TODO:
             // Handle @IntRange annotation when the control flow refinement is implemented
             // typetools/checker-framework#1163
```

Comment:
@kelloggm FYI, I think this can be done after this PR is merged. (Maybe by you maybe someone else.)